things/service/src/main/java/org/eclipse/ditto/things/service/utils/PatchConditionsEvaluator.java

```diff
@@ -0,0 +1,179 @@
+import org.eclipse.ditto.rql.query.filter.QueryFilterCriteriaFactory;
+import org.eclipse.ditto.rql.query.things.ThingPredicateVisitor;
+import org.eclipse.ditto.things.model.Thing;
+import org.eclipse.ditto.things.model.signals.commands.modify.MergeThing;
+
+/**
+ * Utility class for evaluating patch conditions in Thing operations.
+ * <p>
+ * This class provides methods to evaluate RQL-based conditions against existing Things
+ * and filter payloads based on those conditions. It eliminates code duplication between
+ * different command strategies that need to handle patch conditions.
+ * </p>
+ *
+ * @since 3.8.0
+ */
+@Immutable
+public final class PatchConditionsEvaluator {
+
+    private static final TimePlaceholder TIME_PLACEHOLDER = TimePlaceholder.getInstance();
+
+    private PatchConditionsEvaluator() {
+        throw new AssertionError();
+    }
+
+    /**
+     * Evaluates patch conditions for a MergeThing command and filters the merge value accordingly.
+     * This method applies RQL conditions to filter out parts of the merge payload
+     * that should not be applied based on the current state of the Thing.
+     * 
+     * <p>
+     * The patch conditions are provided as a JSON object where each key represents
+     * a JSON pointer path and each value is an RQL expression. If the RQL expression
+     * evaluates to {@code false} against the existing Thing, the corresponding part
+     * of the merge payload at that path will be removed.
+     * </p>
+     *
+     * @param existingThing the current state of the Thing
+     * @param mergeValue the original merge value to be filtered
+     * @param command the MergeThing command containing patch conditions
+     * @return the filtered merge value with parts removed based on failed conditions
+     */
+    public static JsonValue evaluatePatchConditions(final Thing existingThing,
+            final JsonValue mergeValue,
+            final MergeThing command) {
+        
+        final var patchConditionsOpt = command.getPatchConditions();
+        if (patchConditionsOpt.isEmpty()) {
+            return mergeValue;
+        }
+
+        if (!mergeValue.isObject()) {
+            return mergeValue;
+        }
+
+        final Map<JsonPointer, String> patchConditions = patchConditionsOpt.get();
+        final JsonObject mergeObject = mergeValue.asObject();
+
+        final JsonObjectBuilder adjustedPayloadBuilder = mergeObject.toBuilder();
+
+        for (final Map.Entry<JsonPointer, String> entry : patchConditions.entrySet()) {
+            final JsonPointer conditionPath = entry.getKey();
+            final String conditionExpression = entry.getValue();
+
+            final boolean conditionMatches = evaluateCondition(existingThing, conditionExpression, command.getDittoHeaders());
+            final boolean containsResource = mergeObject.getValue(conditionPath).isPresent();
+
+            if (!conditionMatches && containsResource) {
```

Comment:
@hu-ahmed small optimisation: there is no need to evaluate the condition (which is a more expensive operation than checking for presence of the resource) if `containsResource` is already `false` ..